### ISSUE TITLE
Running Dokken as non-root User

### ISSUE THREAD

**Issue by pantocrator27** [someara/kitchen-dokken#100]
Hello.  I am running a Jenkins 2 instance with docker based builds that are running as a `jenkins` user.  I am able to pull the platform based images but the chef image seems to run as root.  The only way I can make this work is to mount the `$HOME/.dokken` directory inside of the container that is running as the jenkins user but it appears as if this directory is created as a root user.  The build fails because there is a permissions mismatch between the containers.  Is there a way to control the user that the `$HOME/.dokken` is created as?

**Comment by pantocrator27** [comment]
the trick was setting permissions on `$HOME/.dokken`

**Comment by someara** [comment]
Ha. I'm glad you figured this out. I was going to set up a Jenkins cluster to investigate!

-s

**Comment by pantocrator27** [comment]
In re-reading this, I could have been a bit clearer so I am going to recap the issue for posterity ...

I am running a Jenkins 2 master & slave.  The slaves only contain Jenkins & Docker with the expectation that all builds on the slave will be done in Docker containers.  To facilitate communication between the host and the slave, I have the Jenkins UID pegged at 10000 and the Docker GID pegged at 11000.  Inside of my build image, it has a Jenkins user created with its UID & GID pegged at 10000 in the container.  Furthermore, the image itself is running as the Jenkins user.  

When I first tried testing with dokken, I had a Jenkinsfile in which I was starting the container with the following args to docker:

```
--user 10000:11000 --volume /var/run/docker.sock:/var/run/docker.sock --volume /home/jenkins/.dokken:/home/jenkins/.dokken
```

The trick was that `$HOME/.dokken` was mounted inside of my container as `root` when my container was running as `jenkins`.  So I had to install `sudo` inside of my build container so that before doing the dokken testing, I changed the permissions of that folder and all was right with the world.  

Thanks!
